### ISSUE TITLE
Additional APIs / improvements to parsing

### ISSUE THREAD

**Issue by rcoup** [getsentry/raven-node#82]
Currently it's actually pretty messy to add a contextual description to an error, or add parsed request info to messages or errors outside of the middleware. 

``` javascript
function someView(req, res) {
  doStuff(function(err, data) {
    if (err) {
      parsers.parseError(err, parsers.parseRequest(req), function(kwargs) {
        var result = client.captureMessage("error/context description", kwargs);
        res.send(400, "Badness: " + client.getIdent(result));
      });
    } else {
      res.send(200, data);
    }
  }
}
```

Seems we could make it a lot simpler pretty easily:
1. add an optional description to captureError():

```
client.captureError(Error error[, String description][, Object options][, Function callback])
```
1. automatically look for a `request` (or `req`) in any `options` passed to captureMessage() and captureError(). If found, run parseRequest() on it.

The above then becomes

``` javascript
function someView(req, res) {
  doStuff(function(err, data) {
    if (err) {
      client.captureError(err, "context description", {request: req}, function(result) {
        res.send(400, "Badness: " + client.getIdent(result));
      });
    } else {
      res.send(200, data);
    }
  }
}
```

and passing request information along with a captureMessage() call becomes:

``` javascript
client.captureMessage("some message", {request:req});
```

Thoughts on the concept?

**Comment by benvinegar** [comment]
@MaxBittker – this is an old issue, but I've seen a bunch of related issues more recently. I'm in favor of us automatically attempting parsing a `request` property if we find it as an option to `captureException`/`captureMessage`.

**Comment by LewisJEllis** [comment]
Couple more related areas we can grab the request more automatically: in `process` if there's a `context.request` value and no `kwargs.request`, we might as well read the request from the context. Then in the `requestHandler` middleware we can add the request to the context; this way errors captured after the `requestHandler` middleware but *not* by the `errorHandler` middleware (i.e. if someone does a manual `captureException`) will still have the request associated, when they currently do not.

I think there are a couple more similar "could be slightly more automatic with context association for semi-manual capture cases" spots but I don't remember them exactly right now - will make sure to shore up any gaps.

**Comment by LewisJEllis** [comment]
Resolved in #310
